compiler/src/dotty/tools/dotc/transform/SyntheticMembers.scala

```diff
@@ -389,23 +401,24 @@ class SyntheticMembers(thisPhase: DenotTransformer) {
   /** If this the class backing a serializable singleton enum value with base class `MyEnum`,
    *  and not deriving from `java.lang.Enum` add the method:
    *
-   *      private def writeReplace(): AnyRef =
-   *        new scala.runtime.EnumValueSerializationProxy(classOf[MyEnum], this.ordinal)
+   *      private def readResolve(): AnyRef =
+   *        MyEnum.values(this.ordinal)
    *
    *  unless an implementation already exists, otherwise do nothing.
    */
    def serializableEnumValueMethod(clazz: ClassSymbol)(using Context): List[Tree] =
     if clazz.isEnumValueImplementation
       && !clazz.derivesFrom(defn.JavaEnumClass)
       && clazz.isSerializable
-      && !hasWriteReplace(clazz)
+      && !hasReadResolve(clazz)
     then
       List(
-        DefDef(writeReplaceDef(clazz),
-          _ => New(defn.EnumValueSerializationProxyClass.typeRef,
-                   defn.EnumValueSerializationProxyConstructor,
-                   List(Literal(Constant(clazz.classParents.head)), This(clazz).select(nme.ordinal).ensureApplied)))
-          .withSpan(ctx.owner.span.focus))
+        DefDef(readResolveDef(clazz),
```

Comment:
I think the advantage of writeReplace over readResolve is that it makes serialization cheaper since we don't serialize any potential field defined in the enum

Reply:
that's true, how about where in the EnumValueSerializationProxy where I was caching the .values array to avoid reflection each time, keep that as it is?

Reply:
Not sure. Are all enums with `.values` instances of java.lang.Enum? If so you could access .values without reflection

Reply:
the jvm does its own thing for `java.lang.Enum` so `EnumValueSerializationProxy` doesn't get used for them.

Reply:
I think one option could be the addition of a `byOrdinal` mixin trait for the companion and serialize that with the ordinal and that would be fast

Reply:
Serialization/deserialization isn't a very fast operation anyway, so I wouldn't worry too much about it.